src/main/java/com/dataguise/saas/service/ConnectionService.java

```diff
@@ -0,0 +1,450 @@
+                HttpStatus.BAD_REQUEST);
+        }
+        return connectionId;
+    }
+
+    /**
+     * Update a connection.
+     *
+     * @param connectionDTO the updated connection
+     * @return connectionId
+     */
+    public Integer editConnection(Integer connectionId, ConnectionDTO connectionDTO)
+        throws InternalServerException, BadRequestException, BadGatewayException {
+        DgConnection updatedConnection = connectionDTO.getConnection();
+        Boolean defaultSchedule = connectionDTO.getAutoSchedule();
+        if (connectionId == null) {
+            throw new BadRequestException(
+                "Connection " + ErrorConstants.ERROR_MISSING_PATH_VARIABLE.getErrorMessage(),
+                HttpStatus.BAD_REQUEST);
+        }
+        try {
+            if (updatedConnection != null) {
+                if (updatedConnection.getConnectionId() != null) {
+                    logger.info("Connection already exists.");
+                    /* check if user changed the password or not.*/
+                    DgConnection currDbConnection = getConnections(
+                        Optional.of(updatedConnection.getConnectionId()))
+                        .get(0).getConnection();
+                    if (currDbConnection.getPassword() != null
+                        && updatedConnection.getPassword() != null) {
+                        if (currDbConnection.getPassword()
+                            .equals(updatedConnection.getPassword())) {
+                            // set password to "" if the password isn't changed.
+                            // Don't send encrypted password from saas ui to re encrypt.
+                            updatedConnection.setPassword("");
+                        }
+                    }
+                }
+                updatedConnection.setModule(connectionDTO.getConnection().getModule());
+                connectionId = restDriver.editConnection(connectionId, updatedConnection);
+            }
+        } catch (DgDrvException e) {
+            logger.error("Edit connection failed: " + e);
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("Connection edited with id: " + connectionId);
+        try {
+            if (defaultSchedule != null && defaultSchedule == true) {
+                getScheduleInfo(connectionId);
+                SchedulerDTO schedule = getScheduleInfo(connectionId);
+                logger.info("Creating a schedule for connection with id: " + connectionId);
+                Boolean scheduleStatus = createDefaultDetectionSchedule(schedule,
+                    connectionDTO.getConnection().getModule());
+                if (scheduleStatus == true) {
+                    logger.info("Default Schedule task for connection id " + connectionId
+                        + " completed successfully.");
+                } else {
+                    logger.error(
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ConnectionService.java L271
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a262257575>*